safekeeper/src/timeline.rs

```diff
@@ -674,6 +681,10 @@ impl Timeline {
     ) -> Result<bool> {
         self.cancel(shared_state);
 
+        // Wait for any concurrent tasks to stop using this timeline, to avoid e.g. attempts
+        // to read deleted files.
+        self.gate.close().await;
+
         // TODO: It's better to wait for s3 offloader termination before
         // removing data from s3. Though since s3 doesn't have transactions it
         // still wouldn't guarantee absense of data after removal.
```

Comment:
Can this comment be removed now? Also, the reference to background activities in the method comment?